src/regmod/models/tobit.py

```diff
@@ -0,0 +1,218 @@
+        -----
+        User input for parameter attribute `inv_link` is ingored and
+        defaults used instead.
+
+        """
+        if not jnp.all(data.obs >= 0):
+            raise ValueError("Tobit model requires non-negative observations.")
+        super().__init__(data, **kwargs)
+
+        # Use JAX inv_link functions
+        for ii, param_name in enumerate(self.param_names):
+            default_link = self.default_param_specs[param_name]["inv_link"]
+            self.params[ii].inv_link = default_link
+
+        # Use JAX data structures
+        self.mat = [jnp.asarray(mat) for mat in self.mat]
+        self.uvec = jnp.asarray(self.uvec)
+        self.gvec = jnp.asarray(self.gvec)
+        self.linear_uvec = jnp.asarray(self.linear_uvec)
+        self.linear_gvec = jnp.asarray(self.linear_gvec)
+        self.linear_umat = jnp.asarray(self.linear_umat)
+        self.linear_gmat = jnp.asarray(self.linear_gmat)
+
+    @partial(jit, static_argnums=(0,))
+    def objective(self, coefs: ArrayLike) -> float:
+        """Get negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        float
+            Negative log likelihood.
+
+        """
+        # self.data.weights included in get_nll_terms
+        obj_param = self.data.trim_weights*self.get_nll_terms(coefs)
+        return obj_param.sum() + self.objective_from_gprior(coefs)
+
+    @partial(jit, static_argnums=(0,))
+    def gradient(self, coefs: ArrayLike) -> DeviceArray:
+        """Get gradient of negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        DeviceArray
+            Gradient of negative log likelihood.
+
+        """
+        return grad(self.objective)(coefs)
+
+    @partial(jit, static_argnums=(0,))
+    def hessian(self, coefs: ArrayLike) -> DeviceArray:
+        """Get hessian of negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        DeviceArray
+            Hessian of negative log likelihood.
+
+        """
+        return hessian(self.objective)(coefs)
+
+    @partial(jit, static_argnums=(0,))
+    def nll(self, params: List[ArrayLike]) -> DeviceArray:
+        """Get terms of negative log likelihood wrt parameters.
+
+        Parameters
+        ----------
+        params : list[array_like]
+            Model parameters.
+
+        Returns
+        -------
+        DeviceArray
+            Terms of negative log likelihood.
+
+        """
+        y = self.data.obs
+        mu = params[0]
+        sigma = params[1]
+        pos_term = jnp.log(sigma) - logpdf((y - mu)/sigma)
+        npos_term = -logcdf(-mu/sigma)
+        return jnp.where(y > 0, pos_term, npos_term)
```

Comment:
Great! @kels271828 I think here we should apply `weights` right?

Reply:
No, the weights are applied in the objective

Reply:
Oh, sorry, the `weights` are applied in the `get_nll_terms`, I also get tripped here. Everything is good.